File: ballast-core/npm/src/domain/events/player-left-game.ts
import { EventBase } from "../../messaging/event-base";
import { Game } from "../models/game";
import { Player } from "../models/player";

export class PlayerLeftGameDomainEvent extends EventBase {

    public static readonly id: string = "PlayerLeftGameDomainEvent";

    public get id() {
        return PlayerLeftGameDomainEvent.id;
    }

    public readonly gameId: string; 
    public readonly player: Player;

    private constructor(eventDateIsoString: string, gameId: string, player: Player) {
        super(eventDateIsoString);
        this.gameId = gameId;
        this.player = player;
    }

    public static fromPlayerInGame(game: Game, player: Player) {
        return new PlayerLeftGameDomainEvent(
            EventBase.getDateIsoString(),
            game.id,
            player
        );
    }

}

File: ballast-core/dotnet/src/application/events/PlayerLeftGame.cs
using System;
using Ballast.Core.Application.Models;
using Ballast.Core.Messaging;

namespace Ballast.Core.Application.Events
{
    public class PlayerLeftGameEvent : EventBase, IApplicationEvent  
    {

        public static string GetId() => nameof(PlayerLeftGameEvent);
        public override string Id => GetId();

        public Guid GameId { get; private set; }
        public PlayerDto Player { get; private set; }

        private PlayerLeftGameEvent() { } // For model binding / deserialization
        private PlayerLeftGameEvent(string eventDateIsoString, Guid gameId, PlayerDto player) : base(eventDateIsoString) 
        {
            GameId = gameId;
            Player = player;
        }

        public static PlayerLeftGameEvent FromPlayerInGame(Guid gameId, PlayerDto player) =>
            new PlayerLeftGameEvent(
                EventBase.GetDateIsoString(),
                gameId, 
                player
            );

    }
}